Dockerfile

```diff
@@ -1,59 +1,81 @@
 # Copyright 2023 The OWASP Coraza contributors
 # SPDX-License-Identifier: Apache-2.0
 
-FROM golang:1.19.1-alpine3.16 AS build
+FROM --platform=$BUILDPLATFORM golang:1.19-alpine3.17 AS builder
 
-# Specify Coreruleset version to download
-ARG CORERULESET_VERSION=v4.0.0-rc1
-ARG CORERULESET_MD5=9140236dc7e941c274e414385824c996
-
-# Change working directory
-WORKDIR /app
-
-RUN \
-    apk add --no-cache \
-    # Install make to build coraza-spoa binary from makefile
-    make \
-    # Download and set up Coreruleset
-    && wget -qO/tmp/coreruleset.tar.gz https://github.com/coreruleset/coreruleset/archive/${CORERULESET_VERSION}.tar.gz \
-    && echo "$CORERULESET_MD5  /tmp/coreruleset.tar.gz" | md5sum -c \
-    && mkdir -p /tmp/coraza-coreruleset \
-    && mkdir -p /etc/coraza-spoa/rules \
-    && tar xzf /tmp/coreruleset.tar.gz --strip-components=1 -C /tmp/coraza-coreruleset \
-    && mv /tmp/coraza-coreruleset/crs-setup.conf.example /etc/coraza-spoa/crs-setup.conf \
-    && mv /tmp/coraza-coreruleset/rules /etc/coraza-spoa \
-    && mv /tmp/coraza-coreruleset/plugins /etc/coraza-spoa \
-    && rm -rf /tmp/*
-
-# Cache Go dependencies
-COPY go.mod go.sum ./
+WORKDIR /build
+COPY . /build
+
+# Download dependencies for all platforms once
 RUN go mod download
 
-# Copy project files into build image
-COPY . .
+ARG TARGETOS
+ARG TARGETARCH
+
+RUN apk add --no-cache make ca-certificates \
+    && update-ca-certificates
+
+RUN --mount=type=cache,target=/root/.cache/go-build \
+    --mount=type=cache,target=/go/pkg \
+    OS=${TARGETOS} ARCH=${TARGETARCH} make
+
+# ---
+FROM alpine:3.17 AS main
+
+ARG TARGETARCH
+
+LABEL org.opencontainers.image.authors="OWASP Coraza Maintainers" \
```

Comment:
Shall we use `The OWASP Coraza contributors` to align with other repos?

Reply:
Changed in latest commit